selfdrive/controls/lib/radar_helpers.py

```diff
@@ -130,14 +130,21 @@ def get_RadarState(self, model_prob=0.0):
       "aLeadTau": float(self.aLeadTau)
     }
 
-  def get_RadarState_from_vision(self, lead_msg, v_ego):
+  def get_RadarState_from_vision(self, lead_msg, v_ego, vision_v_ego):
+    # Learn vision model velocity error to correct vlead
+    if v_ego > 0 and vision_v_ego > 0:
+      vision_velocity_factor = v_ego / vision_v_ego
+      corrected_v_lead = lead_msg.v[0] * vision_velocity_factor
+      #if no v_ego or vision_v_ego set, fallback to uncalibrated values
+    else: 
+      corrected_v_lead = lead_msg.v[0]
```

Comment:
I would set v_lead above before the if statement, then override if possible

Reply:
How does it look now? I understand that this is cleaner